src/app/clusters/tls-client-management-server/TlsClientManagementCluster.cpp

```diff
@@ -126,110 +145,135 @@ TlsClientManagementCluster::EncodeProvisionedEndpoints(EndpointId matterEndpoint
-                                               TlsClientManagement::Attributes::ProvisionedEndpoints::Id);
-    }
-    else
-    {
-        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        ConcreteCommandPath responsePath(mPath.mEndpointId, TlsClientManagement::Id, Commands::ProvisionEndpoint::Id);
+        commandHandler.AddResponse(responsePath, response);
+        NotifyAttributeChanged(TlsClientManagement::Attributes::ProvisionedEndpoints::Id);
+        return DataModel::ActionReturnStatus(Status::Success);
     }
+
+    return DataModel::ActionReturnStatus(status);
 }
 
-void TlsClientManagementCluster::HandleFindEndpoint(HandlerContext & ctx, const Commands::FindEndpoint::DecodableType & req)
+std::optional<DataModel::ActionReturnStatus>
+TlsClientManagementCluster::HandleFindEndpoint(CommandHandler & commandHandler, const Commands::FindEndpoint::DecodableType & req)
 {
     ChipLogDetail(Zcl, "TlsClientManagement: FindEndpoint");
 
-    VerifyOrReturn(req.endpointID <= kMaxTlsEndpointId, ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+    if (req.endpointID > kMaxTlsEndpointId)
+    {
+        return DataModel::ActionReturnStatus(Status::ConstraintError);
+    }
+
+    auto fabric     = commandHandler.GetAccessingFabricIndex();
+    auto endpointId = mPath.mEndpointId;
 
     CHIP_ERROR result =
-        mDelegate.FindProvisionedEndpointByID(ctx.mRequestPath.mEndpointId, ctx.mCommandHandler.GetAccessingFabricIndex(),
-                                              req.endpointID, [&](auto & endpoint) -> CHIP_ERROR {
-                                                  Commands::FindEndpointResponse::Type response;
-                                                  response.endpoint = endpoint;
-                                                  ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
-                                                  return CHIP_NO_ERROR;
-                                              });
+        mDelegate.FindProvisionedEndpointByID(endpointId, fabric, req.endpointID, [&](auto & endpoint) -> CHIP_ERROR {
+            Commands::FindEndpointResponse::Type response;
+            response.endpoint = endpoint;
+            ConcreteCommandPath responsePath(mPath.mEndpointId, TlsClientManagement::Id, Commands::FindEndpoint::Id);
+            commandHandler.AddResponse(responsePath, response);
+            return CHIP_NO_ERROR;
+        });
 
     if (result == CHIP_ERROR_NOT_FOUND)
     {
-        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return DataModel::ActionReturnStatus(Status::NotFound);
     }
-    else if (result != CHIP_NO_ERROR)
+
+    if (result != CHIP_NO_ERROR)
     {
-        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+        return DataModel::ActionReturnStatus(Status::Failure);
     }
+
+    return DataModel::ActionReturnStatus(Status::Success);
```

Comment:
```suggestion
    return std::nullopt;
```
because you AddResponse above.